doc/article/en-US/templating-html-behaviors-introduction.md

```diff
@@ -297,7 +297,232 @@ The `@bindable` decorator isn't used when doing single value binding with a Cust
+
+<code-listing heading="video-player${context.language.fileExtension}">
+  <source-code lang="ES 2015">
+    export class VideoPlayer {
+      @bindable({
+        coerce(val) {
+          // Convert any truthy value or an empty string to true
+          if (val || val === '') {
+            return true;
+          }
+          // The rest to false
+          return false;
+        }
+      })
+      playing
+    }
+  </source-code>
+</code-listing>
+
+#### Extending typed bindable / Modifying built-in typed bindable
+
+From examples above, we see that `coerce` is how we tell a `bindable` decorator how to convert incoming value. All `coerce`s are registered at export `coerceFunctions` of the `aurelia-binding`. So to modify a built-in coerce:
+
+<code-listing heading="extend-coerce">
+  <source-code lang="ES 2015">
+    // import from 'aurelia-binding' if you are writing a plugin
+    // import from 'aurelia-framework' if you are writing an application
+    import {coerceFunctions} from 'aurelia-framework';
+
+    // Modify built-in coerce:
+    // We want to treat null and undefined as empty string
+    coerceFunctions.string = function(incomingValue) {
+      return incomingValue === null || incomingValue === undefined ? '' : incomingValue.toString();
+    }
+  </source-code>
+</code-listing>
+
+To add your own `coerce`:
+
+<code-listing heading="extend-coerce">
+  <source-code lang="ES 2015">
+    // import from 'aurelia-binding' if you are writing a plugin
+    // import from 'aurelia-framework' if you are writing an application
+    import {coerceFunctions} from 'aurelia-framework';
+
+    // We will add coerce for a point, which supposed to have x and y
+    // Assume all incoming values are strings that look like this '5.1 6.2'
+    coerceFunctions.point = function(incomingValue) {
+      return incomingValue.split(' ').map(parseFloat);
+    };
+  </source-code>
+</code-listing>
+
+#### Usage with metadata for Typescript
+
+Typescript compiler has an option emit class fields with their types in metadata. `bindable` decorator can work with this via `usePropertyType` function, which is a property of `bindable` decorator:
+
+<code-listing heading="extend-coerce">
+  <source-code lang="ES 2015">
+    bindable.usePropertyType(true);
+  </source-code>
+</code-listing>
+
+After doing this, all of the type in metadata emitted by Typescript will be used to resolve to its equivalent property in `coerceFunctions` above. By default, there are 4 mappings:
+
+  * `Number` to `'number'`
+  * `String` to `'string'`
+  * `Boolean` to `'boolean'`
+  * `Date` to `'date'`
+
+You can extend / modify this via export `coerceFunctionMap` of `aurelia-binding`:
+
+<code-listing heading="extend-coerce">
+  <source-code lang="TypeScript">
+    // import from 'aurelia-binding' if you are writing a plugin
```

Comment:
is it explained somewhere else, why plugins should import things from original package instead of `aurelia-framework` that exports things from other packages? I guess they both work, but 'aurelia-binding' should be preferred to avoid excessive dependencies (and to be tree-shaking friendly)?